demonstrations/tutorial_qpe.py

```diff
@@ -0,0 +1,294 @@
+r"""Intro to Quantum Phase Estimation
+=============================================================
+
+The Quantum Phase Estimation (QPE) algorithm is one of the most important tools in quantum
+computing. Maybe **the** most important. It solves a deceptively simple task: given an eigenstate of a
+unitary operator, find its eigenvalue. This demo explains the basics of the QPE algorithm.
+After reading it, you will be able to understand
+the algorithm and how to implement it in PennyLane.
+
+.. figure:: ../_static/demonstration_assets/qpe/socialthumbnail_large_Quantum_Phase_Estimation_2023-11-27.png
+    :align: center
+    :width: 50%
+
+
+
+Quantum phase estimation
+------------------------
+
+Let's define the problem more carefully. We are given a unitary
+operator :math:`U` and one of its eigenstates :math:`|\psi \rangle`. The operator is unitary,
+so we can write:
+
+.. math::
+    U |\psi \rangle = e^{i \phi} |\psi \rangle,
+
+where :math:`\phi` is the *phase* of the eigenvalue. The goal is to estimate :math:`\phi`,
+hence the name phase estimation. Our challenge is to design a quantum algorithm to solve this problem.
+How would that work?
```

Comment:
Optional suggestion but maybe nice!

```suggestion
where :math:`\phi` is the *phase* of the eigenvalue (remember, unitaries have eigenvalues with an 
absolute value of 1). The goal is to estimate :math:`\phi`,
hence the name *phase estimation*. Our challenge is to design a quantum algorithm to solve this problem.
How would that work?
```